pkgs/top-level/stage.nix

```diff
@@ -292,10 +292,20 @@ let
           pkgsExtraHardening = super';
           stdenv = super'.withDefaultHardeningFlags (
             super'.stdenv.cc.defaultHardeningFlags ++ [
+              "shadowstack"
               "stackclashprotection"
               "trivialautovarinit"
             ]
           ) super'.stdenv;
+          glibc = super'.glibc.override rec {
+            enableCET = if self'.stdenv.hostPlatform.isx86_64 then "permissive" else false;
+            enableCETRuntimeDefault = enableCET != false;
+          };
+        } // lib.optionalAttrs (with super'.stdenv.hostPlatform; isx86_64 && isLinux) {
+          # causes shadowstack disablement
+          pcre = super'.pcre.override { enableJit = false; };
+          pcre-cpp = super'.pcre-cpp.override { enableJit = false; };
+          pcre16 = super'.pcre16.override { enableJit = false; };
```

Comment:
These are the first packages that `pkgsExtraHardening` is making an explicit domain‐specific hardening vs. functionality/performance trade‐off for, rather than just blanket‐defaulting a hardening switch on and letting individual packages decide what to do, right?

I’m not opposed necessarily, just want to check that this is something new that’s happening. I do worry a little about bit rot as package options drift without anyone checking `pkgsExtraHardening`. And it might be hard to make judgement calls in future when you’re faced with whether to turn all JITs ever off for security reasons.

Reply:
Not sure if this answers the question, but here I think is the starting point that lead to this: https://github.com/NixOS/nixpkgs/pull/326819#issuecomment-2238442118 . TLDR: legacy PCRE JIT is broken with shadow stack, so either JIT or shadow stack have to be disabled, and the latter has cascade effects.

Reply:
Right, okay. The fact that it would require disabling it for all the downstream packages makes this make sense to me.

Reply:
Yes, this is the first time we're overriding an actual package.